packages/react-native-web/src/exports/Modal/ModalAnimation.js

```diff
@@ -45,7 +45,9 @@ function ModalAnimation(props: ModalAnimationProps) {
   } = props;
 
   const [isRendering, setIsRendering] = useState(false);
-  const wasVisible = useRef(false);
+
+  // Undefined by default so the initial `onDismiss` callback is fired
```

Comment:
This doesn't seem right to me. `onDismiss` shouldn't be getting called on initial render but it currently is

Reply:
That's.. true.

Looking at the [`react-native` implementation](https://github.com/facebook/react-native/blob/master/Libraries/Modal/Modal.js#L166) `onDismiss` is ONLY called when the modal is being unmounted - not when the modal is being hidden.

Does my interpretation align with yours on this?

Reply:
My confusion is in that https://reactnative.dev/docs/modal.html#ondismiss doesn't align with the code.

Reply:
Through testing on android I'm finding the behavior is this as well.  Perhaps the react docs should be updated.

Reply:
Ok - I've also looked at the last known working code for the dismissal in 0.62 to confirm the behavior matches your suggestion (it does)

while the code that [wraps firing off the onDismiss looks somewhat similar to `onShow`](https://github.com/facebook/react-native/blob/d0871d0a9a373e1d3ac35da46c85c0d0e793116d/React/Views/RCTModalHostView.m#L195)

It's [very different from how it's called in the manager](https://github.com/facebook/react-native/blob/v0.61.5/React/Views/RCTModalHostViewManager.m#L82)

So while `onShow` fires if the modal starts visible, `onDismiss` won't if it starts not-visible.